### PR TITLE
Add release internal workflow

### PR DESCRIPTION
**Pull request by VersusFacit**
### Problem

We need a workflow that calls the internal build release workflow [found here](https://github.com/dbt-labs/dbt-release/pull/107).

### Solution

This PR adds the file and documents the what/how/why. 

image

Proof of calling unit/integration tests and build completing.

### Checklist

- [X] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [X] I have run this code in development and it appears to resolve the stated issue
- [X] This PR includes tests, or tests are not required/relevant for this PR
- [X] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

**Comment by VersusFacit** [comment]
I also need to change the reusable workflow to point to the main branch when the time comes.

**Review by emmyoop — DISMISSED** [review]
Approving, but same comments as https://github.com/dbt-labs/dbt-redshift/pull/735

File: .github/workflows/release-internal.yml [hunk 1]
```diff
@@ -36,9 +34,6 @@ defaults:
   run:
     shell: "bash"
 
-env:
-  PYTHON_TARGET_VERSION: 3.11
-
 jobs:
   invoke-reusable-workflow:
```
Comment (mikealfare) [inline .github/workflows/release-internal.yml:38]: Note to remember to repoint the workflow below to `main`.

**Review by mikealfare — DISMISSED** [review]

**Review by colin-rogers-dbt — APPROVED** [review]

**Comment by VersusFacit** [comment]
Clean release job ran on latest sha. Out it goes!

**Pull request merged**